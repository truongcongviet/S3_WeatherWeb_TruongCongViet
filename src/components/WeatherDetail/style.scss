.detail-weather {
    height: 45vh;
    .cloudyInfo {
        display: flex;
        justify-content: center;
        align-items: center;
        height: 65%;
        .weatherData {
            width: 250px;
        }
        .cloud-icon {
            font-size: 75px;
        }
    }
    .rainingInfo {
        display: flex;
        justify-content: space-between;
        text-align: center;
        height: 35%;
        li {
            width: 25%;
            list-style: none;
            border-width: 1px;
            border-style: solid;
            display: flex;
            flex-direction: column;
            justify-content: space-between;
            border-image: linear-gradient(to top, white, rgba(0, 0, 0, 0)) 1 100%;
            div{
                font-size: 28px;
            }
            .feel_like{
                margin: auto;
                padding: 5px 5px;
                background-color: var(--success);
                border-radius: 5px;
                text-align: center;
            }
            .dengue {
                width:45px;
                height: 45px;
                background-color: var(--lightcyan);
                border: 2px solid var(--white);
                border-radius: 50%;
                margin: auto;
            }
        }
        
    }
}
