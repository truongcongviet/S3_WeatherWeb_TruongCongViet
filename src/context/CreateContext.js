import { createContext, useCallback, useContext, useReducer, Fragment } from 'react';
import { weatherReducer, initialState } from './reducers/weatherReducer'
const CreateContext = createContext({});

export const useAppContext = () => {
    const { data, setData } = useContext(CreateContext)
    const dispatch = useCallback(
        ({ type, data }) => {
            setData({ data, type })
        },
        [setData]
    )
    return [data, dispatch]
}

export const WeatherProvider = props => {
    const { children } = props;
    const [data, dispatch] = useReducer(weatherReducer, initialState);
    return (
        <CreateContext.Provider value={{ data, setData: dispatch }}>
            <Fragment>
                {children}
            </Fragment>

        </CreateContext.Provider>
    )
}

export default CreateContext;