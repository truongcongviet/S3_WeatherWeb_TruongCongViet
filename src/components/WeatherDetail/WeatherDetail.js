import { Fragment, useEffect } from "react";
import { useAppContext } from "../../context/CreateContext";
import "./style.scss";
const WeatherDetail = () => {
    const [{ weather }] = useAppContext();
    console.log('dataweather', weather);
    return (
        <div className="weatherDetail">
            <div className="cloudyInfo">
                <div className="cloud-icon p-2 mb-4">
                    {weather ?
                        <img src={`http://openweathermap.org/img/wn/${weather && weather.current.weather[0].icon}@2x.png`} alt='cloud-icon' />
                        : <i className="bi bi-cloud-minus"></i>}
                </div>
                <div className="weatherData p-2">
                    <div className="h2">{weather && weather.current.weather[0].main}</div>
                    <div className="d-flex h4 justify-content-between">
                        <div className=""><i className="bi bi-thermometer-half"></i> {weather && weather.current.temp} Â°C</div>
                        <div><i className="bi bi-droplet"></i> {weather && weather.current.humidity} %</div>
                    </div>
                </div>
            </div>
            <ul className="rainingInfo p-0">
                <li>
                    <h5>PSI</h5>
                    <div className="feel_like">{weather && weather.current.feels_like}</div>
                    <p>good</p>
                </li>
                <li>
                    <h5>RAIN</h5>
                    <div>{weather && weather.current.rain?weather.current.rain["1h"]:0}</div>
                    <p>mm</p>
                </li>
                <li>
                    <h5>DENGUE</h5>
                    <div className="dengue"></div>
                </li>
                <li>
                    <i className="bi bi-plus-circle h1 text-dark"></i>
                    <p className="text-dark font-weight-bold h4">add</p>
                </li>
            </ul>

        </div>
    )
}
export default WeatherDetail;