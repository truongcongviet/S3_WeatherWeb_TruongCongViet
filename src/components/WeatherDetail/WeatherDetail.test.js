import React from 'react';
import WeatherDetail from './WeatherDetail';
import { mount, configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { WeatherProvider } from '../../context/CreateContext'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const wrapper = mount(<WeatherProvider><WeatherDetail /> </WeatherProvider>);
describe('WeatherHeader Component', () => {
    test("expect to render WeatherHeader component", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const weatherDetailComponent = wrapper.find(".detail-weather");
        expect(weatherDetailComponent.length).toBe(1);;
    });
    
})
