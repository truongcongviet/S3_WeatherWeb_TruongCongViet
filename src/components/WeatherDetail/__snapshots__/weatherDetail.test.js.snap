// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherHeader Component expect to render WeatherHeader component 1`] = `
<WeatherProvider>
  <WeatherDetail>
    <div
      className="detail-weather"
    >
      <div
        className="cloudyInfo"
      >
        <div
          className="cloud-icon p-2 mb-4"
        >
          <i
            className="bi bi-cloud-minus"
          />
        </div>
        <div
          className="weatherData p-2"
        >
          <div
            className="h2"
          />
          <div
            className="d-flex h4 justify-content-between"
          >
            <div
              className=""
            >
              <i
                className="bi bi-thermometer-half"
              />
               
               Â°C
            </div>
            <div>
              <i
                className="bi bi-droplet"
              />
               
               %
            </div>
          </div>
        </div>
      </div>
      <ul
        className="rainingInfo p-0"
      >
        <li>
          <h5>
            PSI
          </h5>
          <div
            className="feel_like"
          />
          <p>
            good
          </p>
        </li>
        <li>
          <h5>
            RAIN
          </h5>
          <div>
            0
          </div>
          <p>
            mm
          </p>
        </li>
        <li>
          <h5>
            DENGUE
          </h5>
          <div
            className="dengue"
          />
        </li>
        <li>
          <i
            className="bi bi-plus-circle h1 text-dark"
          />
          <p
            className="text-dark font-weight-bold h4"
          >
            add
          </p>
        </li>
      </ul>
    </div>
  </WeatherDetail>
   
</WeatherProvider>
`;
